Assets/Project/Modules/Enemies/GeneralEnemyScripts/VFXRelated/EnemyVisuals.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using UnityEngine;
+
+public class EnemyVisuals : MonoBehaviour
+{
+    //This is the base script, any unique additions to these
+    //effects must be done by creating a new unique script
+    //that inherits from this one.
+    
+    [System.Serializable]
+    public class OriginalMeshData
+    {
+        public MeshRenderer _mesh;
+        [HideInInspector] public Material _originalMaterial;
+    }
+
+    [SerializeField] private GeneralEnemyVFXConfig _visualConfig;
+    [SerializeField] [Tooltip("First mesh on the list will be the one that gets 'hurt'")]
+    private List<OriginalMeshData> _originalMeshDatas = new();
+
+    private void Awake()
+    {
+        foreach (var data in _originalMeshDatas)
+        {
+            data._originalMaterial = data._mesh.material;
+        }
+    }
+
+    public virtual void OnHitEffects(float healthCoef)
```

Comment:
Ok this is just nitpicking, and I should have specified it in the naming conventions document, but when there are float parameters that need to be between the ranges of [0,1], I like specifying it in the name (healthCoefPer1, or healthCoef01).

For example, the HealthSystem class implements the following:
public abstract class AValueStat {
        public abstract float GetValuePer1Ratio();
    }